code/modules/xenomorph/xeno_structures.dm

```diff
@@ -1050,3 +1050,39 @@ TUNNEL
 	light_initial_color = LIGHT_COLOR_PURPLE
 	ammo = /datum/ammo/xeno/sticky
 	firerate = 5
+
+
+/obj/structure/xeno/evotower
+	name = "evolution tower"
+	desc = "A sickly outcrop from the ground. It seems to ooze a strange chemical that shimmers and warps the ground around it."
+	icon = 'icons/Xeno/2x2building.dmi'
+	icon_state = "evotower"
+	bound_width = 64
+	bound_height = 64
+	obj_integrity = 600
+	max_integrity = 600
+	///hivenumber of this tower
+	var/hivenumber
+	///boost amt to be added per tower per cycle
+	var/boost_amount = 0.25
+
+/obj/structure/xeno/evotower/Initialize(mapload, hivenum)
+	. = ..()
+	GLOB.hive_datums[hivenum].evotowers += src
+	hivenumber = hivenum
+
+/obj/structure/xeno/evotower/Destroy()
+	GLOB.hive_datums[hivenumber].evotowers -= src
+	return ..()
+
+/obj/structure/xeno/xeno_turret/ex_act(severity)
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			take_damage(700)
+		if(EXPLODE_HEAVY)
+			take_damage(500)
+		if(EXPLODE_LIGHT)
+			take_damage(300)
+
+/obj/structure/xeno/evotower/get_xeno_hivenumber()
```

Comment:
should be with other get_xeno_hivenumber proc

Reply:
done